/*
 * @author Yanzhe Lee [lee.yanzhe@yanzhe.org]
 */
syntax = "proto3";
package com.hitnslab.dnssecurity.deeparcher.api.proto;
option java_package = "com.hitnslab.dnssecurity.deeparcher.api.proto.generated.java";
option java_outer_classname = "MatrixProto";
option optimize_for = SPEED;
option cc_enable_arenas = true;

message DenseMatFloat {
  repeated uint64 dims = 1;
  repeated float values = 10;
}

message CooMatFloat {
  repeated uint64 dims = 1;
  repeated uint64 rows = 2;
  repeated uint64 cols = 3;
  repeated float values = 10;
}

message CsrMatFloat {
  repeated uint64 dims = 1;
  repeated uint64 row_offsets = 2;
  repeated uint64 col_indices = 3;
  repeated float values = 10;
}

message CscMatFloat {
  repeated uint64 dims = 1;
  repeated uint64 col_offsets = 2;
  repeated uint64 row_indices = 3;
  repeated float values = 10;
}

message DenseMatInt64 {
  repeated uint64 dims = 1;
  repeated int64 values = 10;
}

message CooMatInt64 {
  repeated uint64 dims = 1;
  repeated uint64 rows = 2;
  repeated uint64 cols = 3;
  repeated int64 values = 10;
}

message CsrMatInt64 {
  repeated uint64 dims = 1;
  repeated uint64 row_offsets = 2;
  repeated uint64 col_indices = 3;
  repeated int64 values = 10;
}

message CscMatInt64 {
  repeated uint64 dims = 1;
  repeated uint64 col_offsets = 2;
  repeated uint64 row_indices = 3;
  repeated int64 values = 10;
}
